src/MultiTypeSymbolTable.chpl

```diff
@@ -90,9 +94,8 @@ module MultiTypeSymbolTable
 
             tab.addOrReplace(name, entry);
             entry.setName(name);
-            // When we retrieve from table, it comes back as AbstractSymEntry so we need to cast it
-            // back to the original type. Since we know it already we can skip isAssignableTo check
-            return (tab[name]:borrowed GenSymEntry).toSymEntry(t, N);
+
+            return entry :borrowed :unmanaged :borrowed; // suppress lifetime checking
```

Comment:
Why is this needed?

Reply:
Good question. First, this avoids a lookup in the symtable for the thing we just put there. Second, it works around the lifetime checker that would flag it an error if we just returned `entry :borrowed`.

Reply:
Thanks for explaining that. Resolved.